frr-tools/cp-tool/cp-tool.go

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func CopyFile(srcFile string, destFile string) error {
+	srcFile = filepath.Clean(srcFile)
+	destFile = filepath.Clean(destFile)
+	src, err := os.Open(srcFile)
+	if err != nil {
+		err = fmt.Errorf("failed to open source file %s: %w", srcFile, err)
+		return err
+	}
+	defer src.Close()
+
+	srcStat, err := os.Stat(srcFile)
+	if err != nil {
+		err = fmt.Errorf("failed to check stats for %s: %w", srcFile, err)
+		return err
+	}
+
+	srcPerm := srcStat.Mode().Perm()
+	fmt.Printf("File permission before copying %v \n", srcPerm)
+
+	dst, err := os.Create(destFile)
+	if err != nil {
+		err = fmt.Errorf("failed to create destination file: %s: %w", destFile, err)
+		return err
+	}
+	defer dst.Close()
+
+	_, err = io.Copy(dst, src)
+	if err != nil {
+		err = fmt.Errorf("failed to copy %s to %s: %w", srcFile, destFile, err)
+		return err
+	}
+
+	err = os.Chmod(destFile, srcStat.Mode())
+	if err != nil {
+		err = fmt.Errorf("failed to apply permission on %s", destFile)
+		return err
+	}
+
+	dstStat, err := os.Stat(destFile)
+	if err != nil {
+		err = fmt.Errorf("failed to check stats for %s: %w", destFile, err)
+		return err
+	}
+
+	dstPerm := dstStat.Mode().Perm()
+	fmt.Printf("File permission After copying %v \n", dstPerm)
+
+	return nil
+}
+
+func main() {
+	if len(os.Args) != 3 {
+		fmt.Printf("Please provide two command line arguments.")
+		return
+	}
+
+	sourceFile, destinationFile := os.Args[1], os.Args[2]
+	sourceFile = filepath.Clean(sourceFile)
+	destinationFile = filepath.Clean(destinationFile)
```

Comment:
small nit: this is probably redundant with the one happening inside the function

Reply:
removed the redundant part.